fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/query/visitors/ExpressionVisitor.java

```diff
@@ -549,6 +549,23 @@ public Expression visitBinaryComparisonPredicate(@Nonnull RelationalParser.Binar
         return getDelegate().resolveFunction(ctx.comparisonOperator().getText(), left, right);
     }
 
+    @Nonnull
+    @Override
+    public Expression visitBetweenComparisonPredicate(@Nonnull RelationalParser.BetweenComparisonPredicateContext ctx) {
+        final var operand = Assert.castUnchecked(ctx.operand.accept(this), Expression.class);
+        final var left = Assert.castUnchecked(ctx.left.accept(this), Expression.class);
+        final var right = Assert.castUnchecked(ctx.right.accept(this), Expression.class);
+        if (ctx.NOT() == null) {
+            return getDelegate().resolveFunction("and",
+                    getDelegate().resolveFunction("<=", left, operand),
+                    getDelegate().resolveFunction("<=", operand, right));
+        } else {
+            return getDelegate().resolveFunction("or",
+                    getDelegate().resolveFunction("<", operand, left),
+                    getDelegate().resolveFunction(">", operand, right));
```

Comment:
it feels like this is something for the optimizer to do, for example, if we have a filtered index on something like `where x between 40 and 100`, and the query is something like `where not x between 40 and 100`, I guess the optimizer might have a better chance of matching that index and use it to perform an anti-join agains the base table for example.

I am not so sure though this is a good enough argument for not doing such pre-mature rewrite.